### PR TITLE
[MPQEditor] Implement 'dispose' and 'onViewMessage'

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit implements 'dispose' and 'onViewMessage' for manual mixed precision quantization task.

Its correctness is tested in https://github.com/Samsung/ONE-vscode/pull/1530

Fresh draft: https://github.com/Samsung/ONE-vscode/pull/1530
Full draft: https://github.com/Samsung/ONE-vscode/pull/1505
Related: https://github.com/Samsung/ONE-vscode/issues/1491

ONE-vscode-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

File: src/MPQEditor/MPQCircleSelector.ts [hunk 1]
```diff
@@ -213,14 +214,49 @@ export class MPQSelectionPanel
...
   }
 
   /**
    * CircleGraphEvent interface implementations
    */
-  public onViewMessage(_message: any) {
-    // TODO
+  public onViewMessage(message: any) {
+    switch (message.command) {
+      case MessageDefs.selection:
```
Comment (dayo09) [inline src/MPQEditor/MPQCircleSelector.ts:248]: I've got some UI feedbacks. If it's not easy to handle on this pr, you may consider it later.

image

On #1505 draft, 
- when I click on 'add' node, MPQEditor seems to adds the node to the mpq editor slot. 
- when I click on 'Place holder's or 'Add 2' node, as they are not the `mpq` target, they shows the property view on the right side of the circle editor.

I think that 'click' event must result in the same behavior. 
We can choose either to open the property view same to 'add' operation or not to open the property view to other operations.

**Comment by dayo09** [comment]
Hi, this is another UI feedback.
It's not related to this PR, but rather related to the draft #1505.

image

It seems that : In the (1) area, it decides the default quantization configuration and (2) area specifies the per-node(layer) quantization configuration.

At first look, I didn't understand what the (1) area would do. I propose to add 'Set default quantization' 'Set specific layer's quantization' in each areas.

File: src/MPQEditor/MPQCircleSelector.ts (hunk 1 above)
Comment (stamalakhov) [reply src/MPQEditor/MPQCircleSelector.ts:248]: @dayo09 
Thank you. I'll try to fix it in the draft at first.

**Comment by stamalakhov** [comment]
> Hi, this is another UI feedback. It's not related to this PR, but rather related to the draft #1505.
> 
> image
> 
> It seems that : In the (1) area, it decides the default quantization configuration and (2) area specifies the per-node(layer) quantization configuration.
> 
> At first look, I didn't understand what the (1) area would do. From my point of view, I propose to add 'Set default quantization' 'Set specific layer's quantization' in each areas.

@dayo09 
Thank you. I'll try to fix it in the draft at first.

File: src/MPQEditor/MPQCircleSelector.ts (hunk 1 above)
Comment (stamalakhov) [reply src/MPQEditor/MPQCircleSelector.ts:248]: @dayo09 @seanshpark 
The same behaviour in the netron:
 click on input/output nodes shows `Model Properties`, while clicking on the internal nodes show `Nodes Properties`.
For MPQ draft clicking internal nodes results in their selection for editing, while for input/output nodes still shows `Model Properties` to observe input/output size and theirs type.
To handle this issue:
1. We can hide input/output nodes
2. We can disable clicking input/output nodes

But still there may be some issues that some users are used to observe input/output size.
IMHO.

**Comment by stamalakhov** [comment]
> It seems that : In the (1) area, it decides the default quantization configuration and (2) area specifies the per-node(layer) quantization configuration.

@dayo09 
To fix layout it needs to edit media files, so may be we can handle it in another PR?

**Comment by dayo09** [comment]
@stamalakhov  

> To fix layout it needs to edit media files, so may be we can handle it in another PR?

Sure :-D Thanks a lot :-D

File: src/MPQEditor/MPQCircleSelector.ts (hunk 1 above)
Comment (dayo09) [reply src/MPQEditor/MPQCircleSelector.ts:248]: Or 3. We can show `Properties` too, when clicking on the internal nodes(`ex. Add`).

Comment (dayo09) [reply src/MPQEditor/MPQCircleSelector.ts:248]: I will approve this PR, let's go on about it on the other issue or PR. Thanks for listening to me.!

**Review by dayo09 — APPROVED** [review]

File: src/MPQEditor/MPQCircleSelector.ts (hunk 1 above)
Comment (stamalakhov) [reply src/MPQEditor/MPQCircleSelector.ts:248]: > I will approve this PR, let's go on about it on the other issue or PR. Thanks for listening to me.!

Thank you very much!

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**